ghcide/src/Development/IDE/Core/WorkerThread.hs

```diff
@@ -0,0 +1,46 @@
+module Development.IDE.Core.WorkerThread
+    (withWorkerQueue, blockRunInThread)
+ where
+
+import           Control.Concurrent.Async
+import           Control.Concurrent.STM
+import           Control.Concurrent.Strict (newBarrier, signalBarrier,
+                                            waitBarrier)
+import           Control.Monad             (forever)
+import           Control.Monad.Cont        (ContT (ContT))
+
+-- Note [Serializing runs in separate thread]
```

Comment:
I do recommend using multiline comments for stuff like this, it's nicer!

Reply:
good idea done